tests/test_remotesparkmagics.py

```diff
@@ -0,0 +1,94 @@
+from nose.tools import raises, with_setup, assert_equals
+from remotespark.RemoteSparkMagics import RemoteSparkMagics
+from mock import MagicMock
+
+ip = get_ipython()
+magic = None
+
+def _setup():
+	global magic
+	magic = RemoteSparkMagics(shell=ip)
+	ip.register_magics(magic)
+
+def _teardown():
+	pass
+
+@with_setup(_setup, _teardown)
+def test_info_command():
+	mock_method = MagicMock()
+	magic._print_info = mock_method
+	command = "info"
+
+	ip.run_line_magic("spark", command)
+
+	mock_method.assert_called_with()
+
+@with_setup(_setup, _teardown)
+def test_add_endpoint_command():
+	mock_method = MagicMock()
+	magic.add_endpoint = mock_method
+	command = "add"
+	name = "name"
+	language = "python"
+	connection_string = "url=http://location:port;username=name;password=word"
+	line = " ".join([command, name, language, connection_string])
+
+	ip.run_line_magic("spark", line)
+
+	mock_method.assert_called_with(name, language, connection_string)
+
+
+@with_setup(_setup, _teardown)
+def test_delete_endpoint_command():
+	mock_method = MagicMock()
+	magic.delete_endpoint = mock_method
+	command = "delete"
+	name = "name"
+	line = " ".join([command, name])
+
+	ip.run_line_magic("spark", line)
+
+	mock_method.assert_called_with(name)
+
+@with_setup(_setup, _teardown)
+def test_mode_command():
+	mock_method = MagicMock()
+	magic.log_mode = mock_method
+	command = "mode"
+	mode = "debug"
+
+	line = " ".join([command, mode])
+
+	ip.run_line_magic("spark", line)
+
+	mock_method.assert_called_with(mode)
+
+@with_setup(_setup, _teardown)
+def test_cleanup_command():
+	mock_method = MagicMock()
+	magic.cleanup = mock_method
+	command = "cleanup"
+
+	ip.run_line_magic("spark", command)
+
+	mock_method.assert_called_with()
```

Comment:
I'm showing my C# ways here by suggesting that it's passed via the constructor. The alternative would be to change the client_manager after construction, yes?

I still think it's cleaner to pass a client_manager or a mock of it depending on the situation with the constructor, letting the magic handle its lifetime, and not altering the state of the magics out of band.

What do you think?